src/sentry/web/frontend/accounts.py

```diff
@@ -107,14 +107,16 @@ def recover_confirm(request, user_id, hash, mode="recover"):
             password_hash.delete()
             raise LostPasswordHash.DoesNotExist
         user = password_hash.user
-
     except LostPasswordHash.DoesNotExist:
         return render_to_response(get_template(mode, "failure"), {}, request)
 
+    form = RelocationForm if mode == "relocate" else ChangePasswordRecoverForm
     if request.method == "POST":
-        form = ChangePasswordRecoverForm(request.POST, user=user)
+        form = form(request.POST, user=user)
         if form.is_valid():
             with transaction.atomic(router.db_for_write(User)):
+                if mode == "relocate":
+                    user.username = form.cleaned_data["username"]
```

Comment:
Should a relocation form also change `user.is_unclaimed`?

Reply:
Good question, this seems reasonable to me. @azaslavsky wdyt?